memory/libhugetlbfs.py

```diff
@@ -100,9 +102,33 @@ def setUp(self):
         process.run('patch -p1 < %s' % data_dir + '/' + patch, shell=True)
         build.make(self.srcdir, extra_args='BUILDTYPE=NATIVEONLY')
 
+    def fail_check(self, check):
+
+        log = self.logdir + '/stdout'
+        cmd_fail = "grep 'FAIL:' %s  |head -1| awk '{ print $%s}'" % (
+            log, check)
+        if check == 3:
+            env = "32"
+        else:
+            env = "64"
+
+        fail_check = os.popen(cmd_fail).read()
```

Comment:
We are using `os.popen()`, `process.system()` and `process.run()`, all in the same script. Could you unify them?